### PR TITLE
Feature/docker

### PR DESCRIPTION
**Pull request by domerd**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM ubuntu:latest
+
+RUN apt update
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get -y install tzdata
+
+RUN apt install -y build-essential cmake ninja-build libpcap-dev libglib2.0-dev libgcrypt20-dev libc-ares-dev bison flex liblz4-dev libsmi2-dev libgnutls28-dev libminizip-dev libbrotli-dev libsnappy-dev libzstd-dev libnghttp2-dev lua5.1 luajit libspandsp-dev libxml2-dev liblua5.1-dev libluajit-5.1-dev libkrb5-dev python3-pip python3-setuptools
```
Comment (yehonatanz) [inline Dockerfile]: See this link to learn how to optimize `apt` usage for smaller image size
https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#apt-get

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+COPY . /marine
+
+WORKDIR /marine
+
+RUN mkdir -p build && cd build && cmake -DCMAKE_INSTALL_PREFIX=/usr -GNinja ..
+
+RUN ninja -C build marine
+
+RUN cd build && ninja install
```
Comment (yehonatanz) [inline Dockerfile]: Be consistent and use `ninja -C build install` instead of `cd`-ing.

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    - name: "Build docker image"
+      run: docker build -t marine-core .
+
+
+    - uses: docker/build-push-action@v1
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+        repository: domerd/marine-core
```
Comment (yehonatanz) [inline .github/workflows/docker.yml:31]: Add a TODO or something to change that from `domerd/` to something better when we can.
(Something better = `tomlegkov/` or so slash at all and just `marine-core`)

**Review by yehonatanz — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,46 @@
+FROM ubuntu:latest
+
+RUN apt update
```
Comment (yehonatanz) [inline Dockerfile]: Is this still needed now that you `apt update && ...` on the main `RUN` stage?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get -y install tzdata
```
Comment (yehonatanz) [inline Dockerfile]: what is this

**Review by yehonatanz — CHANGES_REQUESTED** [review]

**Review by yehonatanz — APPROVED** [review]

**Review by shaharyarn — APPROVED** [review]

**Pull request merged**